pkg/common/upgrade/managed.go

```diff
@@ -125,20 +99,29 @@ func TriggerManagedUpgrade(h *helper.H) (*configv1.ClusterVersion, error) {
 
 	// Reschedule the upgrade if flag specified
 	if viper.GetBool(config.Upgrade.ManagedUpgradeRescheduled) {
-		err = updateUpgradeWithProvider(upgradeVersion.String())
+		err = updateUpgradeWithProvider()
 		if err != nil {
-			return cVersion, fmt.Errorf("can't reschedule upgrade: %v", err)
+			return nil, fmt.Errorf("can't reschedule upgrade: %v", err)
 		}
 	}
 
 	// The managed-upgrade-operator won't have updated the CVO version yet, and that's fine.
 	// But let's return what it will look like, for the later 'is it upgraded yet' tests.
-	cUpdate := configv1.Update{
-		Version: upgradeVersion.String(),
+	cUpdate := &configv1.Update{}
+
+	releaseName := viper.GetString(config.Upgrade.ReleaseName)
+	if releaseName != "" {
+		upgradeVersion, err := util.OpenshiftVersionToSemver(releaseName)
+		if err != nil {
+			cUpdate.Version = upgradeVersion.String()
+		}
+	}
+	image := viper.GetString(config.Upgrade.Type)
```

Comment:
Should be `config.Upgrade.Image` here?

Reply:
Thanks, nice catch!!
This is now fixed.